### DISCUSSION TITLE
IPMI over I2C

### DISCUSSION THREAD

**Discussion by asm4** [ipmitool/ipmitool#296] [Q&A]
Hello,

I am looking into trying to run ipmitool over I2C and was wondering if this is supported. I have a Pi connected via I2C to another device that has a BMC. I have the character device `/dev/i2c-1` which I can use to write I2C commands to but I am hoping to be able to send ipmi commands instead. If I connect the Pi and other system both to the same network, I can still send commands over the lanplus interface but I would like to see if I2C is possible.

Most commands I run with ipmitool seem to attempt to open the file `/dev/ipmi-N` which doesn't exist on my raspberry pi. I checked the driver support in the kernel for getting that (https://www.kernel.org/doc/html/latest/driver-api/ipmi.html) and made sure that my kernel has `ipmi_msghandler`, `ipmi_devintf`, and `ipmi_ssif`. But still no luck. I suspect the issue is that I haven't correctly configured the `ipmi_ssif` driver on the kernel command line. Am I on the right path with this? I determined the i2c address with `i2cdetect -y 1` and it shows up as `0x10`. Dumping the file `cat /sys/class/i2c-adapter/i2c-1/name` shows me that my adapter is `bcm2835`. This led me to try to add `ipmb_ssif.addr=0x10 ipmi_ssif.adapter=bcm2835 ipmi_ssif.dbg=1,2,4,8 ipmi_ssif.dbg_probe=1` to the kernel command line. However, I didn't specify anything for `ipmi_ssif.slave_addrs`, `ipmi_ssif.tryacpi`,  or `ipmi_ssif.trydmi` which admittedly could be my problem. I don't know if it could also be just something with the raspberry Pi maybe not having some support for this or something. Any ideas?

I did also looked through some of the other interfaces that ipmitool supports and went down a bit of a rabbit hole with the `ipmb-dev-int` driver (https://www.kernel.org/doc/html/latest/driver-api/ipmb.html) since that also looked promising and originally mistook it for `imb` interface. But it doesn't look like `ipmitool` has that as an interface, is that correct?

I appreciate any insight anyone can provide. Thanks to all who work on ipmitool!

**Comment by AlexanderAmelkin** [comment]
IPMI over I2C is called IPMB. `ipmitool` doesn't directly support that, but please take a look at `ipmi-host` driver by Mellanox. I've never tried it myself and can't guarantee that it will help you, but at the fist glance it looks like it can work with the standard OpenIPMI interface (`ipmitool -I open`):

https://github.com/Mellanox/ipmb-host

**Reply by asm4** [reply]
Thank you kindly! I'll definitely look at that.

**Reply by testeract-tj** [reply]
@asm4 how did this end up for you?  I'm looking into using this as well?
